news/2025-05-20-numpy-2-migration-closure.md

```diff
@@ -0,0 +1,14 @@
+# Upcoming closure of NumPy 2.0 migration
+
+NumPy 2.0 was a big change (the first major version in 15 years). For more than a year, we
+have been migrating feedstocks from NumPy 1.x to NumPy 2.x, and while not every affected
+feedstock has been done, we are planning to conclude the migration at the end of May.
+
+For feedstocks that are not compatible with v2.x yet, this means you will have to add
+
+```yaml
+numpy:
+  - 1.26  # or similar
+```
+
+to your `recipe/conda_build_config.yaml`, and then rerender.
```

Comment:
```suggestion
to your `recipe/conda_build_config.yaml`, and then rerender. The suggested pin of 1.26 means that
your package will only be usable with NumPy 1.26 as it is the last release of NumPy in the 1.x series.
```

Reply:
I think that sounds unnecessarily scary. It's true, but what I want to avoid is that people try 1.22, which won't work for all python versions (3.11, 3.12). In fact, I don't want any python-version-dependence in the example here, because we're going to remove numpy from the python-zip, and so overriding numpy per python version won't be possible anymore.

So realistically, the only choices are 1.25 (first to support 3.12) or 1.26.

Reply:
> So realistically, the only choices are 1.25 (first to support 3.12) or 1.26.

it would be great to have this knowledge in the migration guide. saying exactly this would suffice.

Reply:
I've done that in 4ee087c97f74ec777b88ae04b3b920f9c10dbcbe now